File: learn the basics/hashing/limitedsizehash.cpp
#include <iostream>
#include <vector>
using namespace std;

// Function to count the frequency of numbers in the array
vector<int> frequencycount(const vector<int>& arr) {
    int n = arr.size();
    vector<int> hash_array(n, 0); // Initialize a vector of size n with 0s

    for (int num : arr) {
        if (num >= 1 && num <= n) {
            hash_array[num - 1]++;  // (-1) for 1-based indexing
        }
    }

    return hash_array;
}

int main() {
    vector<int> arr = {4, 5, 6, 7, 2};  // Example input array
    vector<int> result = frequencycount(arr);

    cout << "Frequency of numbers in range 1 to " << arr.size() << ":\n";
    for (int i = 0; i < result.size(); ++i) {
        cout << "Number " << i + 1 << ": " << result[i] << endl;
    }

    return 0;
}

File: learn the basics/hashing/limitedsizehash.py
#You are given an array arr[] containing positive integers.
#The elements in the array arr[] range from 1 to n (where n is the size of the array), 
#and some numbers may be repeated or absent. 
#Your task is to count the frequency of all numbers in the range 1 to n and return an 
#array of size n such that result[i] represents the frequency of the number i (1-based indexing).
def frequencycount(arr):
    n = len(arr)
    hash_array = [0] * n
    for num in arr:
        if 1 <= num <= n:
            hash_array[num - 1] += 1  # (-1) for 1-based indexing
    return hash_array

arr = [4, 5, 6, 7, 2]
result = frequencycount(arr)

print(f"Frequency of numbers in range 1 to {len(arr)}:")
for i in range(len(result)):
    print(f"Number {i + 1}: {result[i]}")
